### ISSUE TITLE
potential bug: Variables in included files override ones in parent file

### ISSUE THREAD

**Issue by pbitty** [go-task/task#1800]
When I include a Taskfile that has `env` and `vars` that overlap the parent taskfile, the parent file's variables get overwritten at parent tasks' scope.  

I am running `task` with the following files:

Taskfile.yml
```yaml
version: '3'

includes:
  child: ./Taskfile.child.yml

env:
  ENV_VAR: root-file

vars:
  VAR: root-file

tasks:
  default: |
    echo ENV_VAR is $ENV_VAR
    echo VAR is {{ .VAR }}
```

Taskfile.child.yml
```yaml
version: '3'

env:
  ENV_VAR: child-file

vars:
  VAR: child-file

# No tasks needed for this example.
```

Expected output:
```
ENV_VAR is root-file
VAR is root-file
```

Actual output:
```
ENV_VAR is child-file
VAR is child-file
```

I believe the code responsible is here: https://github.com/go-task/task/blob/114d5e1404a3b65f488b67d10c872f829f9d8312/taskfile/ast/taskfile.go#L56-L57, where merge the included file's variables into the parent.

Is this intentional behaviour?  I would expect that the parent could provide defaults for the child, or override via `include.*.vars`, but it is surprising to have the child influence values on the parent.

If this is indeed a bug, I'd be happy to contribute a fix.

- Task version: HEAD@[114d5e1](https://github.com/go-task/task/commit/114d5e1404a3b65f488b67d10c872f829f9d8312)
- Operating system:  MacOS 14.4.1
- Experiments enabled: None

**Comment by vmaerten** [comment]
Hello!

Variables are not scoped and we can't change it right now because it would be a breaking changes. We had a discussion about it [here](https://github.com/go-task/task/issues/1660#issuecomment-2115871568)

TLDR: users should avoid has much as possible to use conflicting variable names for now because they are not scoped
